interfaces/builtin/location.go

```diff
@@ -0,0 +1,334 @@
+
+import (
+	"bytes"
+
+	"github.com/ubuntu-core/snappy/interfaces"
+)
+
+var locationPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the location service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# DBus accesses
+#include <abstractions/dbus-strict>
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member={Request,Release}Name
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member=GetConnectionUnix{ProcessID,User}
+    peer=(label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=system
+    name="com.ubuntu.location.Service",
+
+# Allow traffic to/from org.freedesktop.DBus for location service
+dbus (receive, send)
+    bus=system
+    path=/
+    interface=org.freedesktop.DBus*,
+
+dbus (receive, send)
+    bus=system
+    path=/com/ubuntu/location/Service{,/**}
+    interface=org.freedesktop.DBus**
+    peer=(label=unconfined),
+`)
+
+var locationConnectedSlotAppArmor = []byte(`
+# Allow connected clients to interact with the service
+
+# Allow the service to host sessions
+dbus (bind)
+    bus=system
+    name="com.ubuntu.location.Service.Session",
+
+# Allow clients to create a session
+dbus (receive)
+    bus=system
+    path=/com/ubuntu/location/Service
+    interface=com.ubuntu.location.Service
+    member=CreateSessionForCriteria,
+
+# Allow clients to query service properties
+dbus (receive)
+    bus=system
+    path=/com/ubuntu/location/Service
+    interface=org.freedesktop.DBus.Properties
+    member=Get,
+
+# Allow clients to set service properties
+dbus (receive)
+    bus=system
+    path=/com/ubuntu/location/Service
+    interface=org.freedesktop.DBus.Properties
+    member=Set,
+
+# Allow clients to request starting/stopping updates
+dbus (receive)
+    bus=system
+    path=/sessions/*
+    interface=com.ubuntu.location.Service.Session
+    member={Start,Stop}PositionUpdates,
+
+dbus (receive)
+    bus=system
+    path=/sessions/*
+    interface=com.ubuntu.location.Service.Session
+    member={Start,Stop}HeadingUpdates,
+
+dbus (receive)
+    bus=system
+    path=/sessions/*
+    interface=com.ubuntu.location.Service.Session
+    member={Start,Stop}VelocityUpdates,
+
+# Allow the service to send updates to clients
+dbus (send)
+    bus=system
+    path=/sessions/*
+    interface=com.ubuntu.location.Service.Session
+    member=UpdatePosition,
+
+dbus (send)
+    bus=system
+    path=/sessions/*
+    interface=com.ubuntu.location.Service.Session
+    member=UpdateHeading,
+
+dbus (send)
+    bus=system
+    path=/sessions/*
+    interface=com.ubuntu.location.Service.Session
+    member=UpdateVelocity,
+
+dbus (send)
+    bus=system
+    path=/com/ubuntu/location/Service
+    interface=org.freedesktop.DBus.Properties
+    member=PropertiesChanged,
```

Comment:
Nice! Note, you can combing many of these: eg, `member="Update{Heading,Position,Velocity}"`